databricks/koalas/tests/test_ops_on_diff_frames.py

```diff
@@ -886,9 +879,51 @@ def test_dot(self):
         kser = ks.from_pandas(pser)
         pser_other = pd.Series([-450, 20, 12, -30, -250, 15, -320, 100, 3], index=midx)
         kser_other = ks.from_pandas(pser_other)
-
         self.assert_eq(kser.dot(kser_other), pser.dot(pser_other))
 
+        pser = pd.Series([0, 1, 2, 3])
+        kser = ks.from_pandas(pser)
+
+        # DataFrame "other" without Index/MultiIndex as columns
+        pdf = pd.DataFrame([[0, 1], [-2, 3], [4, -5], [6, 7]])
+        kdf = ks.from_pandas(pdf)
+        self.assert_eq(kser.dot(kdf), pser.dot(pdf))
+
+        # DataFrame "other" with Index as columns
+        pdf.columns = pd.Index(["x", "y"])
+        kdf = ks.from_pandas(pdf)
+        self.assert_eq(kser.dot(kdf), pser.dot(pdf))
+        pdf.columns = pd.Index(["x", "y"], name="cols_name")
+        kdf = ks.from_pandas(pdf)
+        self.assert_eq(kser.dot(kdf), pser.dot(pdf))
+
+        pdf = pdf.reindex([1, 0, 2, 3])
+        kdf = ks.from_pandas(pdf)
+        self.assert_eq(kser.dot(kdf), pser.dot(pdf))
+
+        # DataFrame "other" with MultiIndex as columns
+        pdf.columns = pd.MultiIndex.from_tuples([("a", "x"), ("b", "y")])
+        kdf = ks.from_pandas(pdf)
+        self.assert_eq(kser.dot(kdf), pser.dot(pdf))
+        pdf.columns = pd.MultiIndex.from_tuples(
+            [("a", "x"), ("b", "y")], names=["cols_name1", "cols_name2"]
+        )
+        kdf = ks.from_pandas(pdf)
+        self.assert_eq(kser.dot(kdf), pser.dot(pdf))
+
+        pdf = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
+        kdf = ks.from_pandas(pdf)
+
+        self.assert_eq((kdf["b"] * 10).dot(kdf["a"]), (pdf["b"] * 10).dot(pdf["a"]))
+        self.assert_eq((kdf["b"] * 10).dot(kdf), (pdf["b"] * 10).dot(pdf))
+        self.assert_eq((kdf["b"] * 10).dot(kdf + 1), (pdf["b"] * 10).dot(pdf + 1))
```

Comment:
These tests can be in `test_series.py`?

Reply:
My implementation will call resolve_copy for `kdf["b"] * 10` and then combine the resolved with the pass-in, so it might still require compute.ops_on_diff_frames to be True.

Reply:
I don't think we need to use `combine_frames` if `same_anchor(self, other)`.

Reply:
How about:

```diff
diff --git a/databricks/koalas/series.py b/databricks/koalas/series.py
index c2b406f..6ed1a29 100644
--- a/databricks/koalas/series.py
+++ b/databricks/koalas/series.py
@@ -4794,15 +4794,30 @@ class Series(Frame, IndexOpsMixin, Generic[T]):
                     raise ValueError("matrices are not aligned")

                 combined = combine_frames(self.to_frame(), other)
+
+                this_data_spark_column_name = combined["this"]._internal.data_spark_column_names[0]
+                that_data_spark_column_names = combined["that"]._internal.data_spark_column_names
             else:
-                new_this = DataFrame(self._internal.resolved_copy)  # type: DataFrame
-                combined = combine_frames(new_this, other)
+                self_column_label = verify_temp_column_name(other, "__self_column__")
+
+                combined = DataFrame(
+                    self._internal.with_new_columns(
+                        (
+                            [
+                                self._internal.data_spark_columns[0].alias(
+                                    name_like_string(self_column_label)
+                                )
+                            ]
+                            + other._internal.data_spark_columns
+                        ),
+                        column_labels=([self_column_label] + other._internal.column_labels),
+                    ).resolved_copy
+                )
+                this_data_spark_column_name = name_like_string(self_column_label)
+                that_data_spark_column_names = other._internal.data_spark_column_names

             sdf = combined._internal.spark_frame

-            this_data_spark_column_name = combined["this"]._internal.data_spark_column_names[0]
-            that_data_spark_column_names = combined["that"]._internal.data_spark_column_names
-
             sdf = sdf.select(
                 [
                     (scol_for(sdf, this_data_spark_column_name) * scol_for(sdf, scol_name)).alias(
```

Reply:
That's wonderful! Thanks! Updated in 6d92a84.